conda/recipes/libraft_distance/build.sh

```diff
@@ -0,0 +1,9 @@
+if [ -n "$MACOSX_DEPLOYMENT_TARGET" ]; then
+    # C++11 requires 10.9
+    # but cudatoolkit 8 is build for 10.11
+    export MACOSX_DEPLOYMENT_TARGET=10.11
+fi
+
+./build.sh clean libraft -v --allgpuarch --compilelibs
+
+# TODO: Somehow just extract the libraft_distance.so
```

Comment:
@jakirkham, your conda packaging expertise will be useful here.

What we want is for the following separate conda packages:
pyraft
libraft-headers
libraft-nn
libraft-distance

I think the python and header-only packages will be straightforward to produce, but the shared libraries will all be compiled together. Is it possible to compile raft only one and then have separate conda packages for each of the shared libraries which are produced? (In other words, we have a single `build.sh` in `conda/recipes/libraft` but we still produce multiple conda packages.

Reply:
Yes conda-build has support for [split packages]( https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section ). This would have one build step. Then there are install scripts that install the right pieces for each package. Additionally one can specify requirements on the different split packages so the other pieces get installed when needed (for example a common shared library they all need). Hope that helps 🙂